src/coreComponents/codingUtilities/Table.cpp

```diff
@@ -0,0 +1,339 @@
+  }
+  else
+  {
+    for( std::size_t idxColumn = 0; idxColumn < m_columns.size(); ++idxColumn )
+    {
+      integer cellSize = m_columns[idxColumn].m_maxStringSize.length();
+      if( idxColumn == 0 )
+      {
+        sectionSeparator +=  GEOS_FMT( "+{:-<{}}", "", ( cellSize + borderMargin ));
+      }
+      else if( idxColumn == (m_columns.size() - 1))
+      {
+        sectionSeparator += GEOS_FMT( "{:-^{}}", "+", columnMargin );
+        sectionSeparator += GEOS_FMT( "{:->{}}", "+", ( cellSize + borderMargin + 1 ) );
+      }
+      else
+      {
+        sectionSeparator += GEOS_FMT( "{:-^{}}", "+", columnMargin );
+        sectionSeparator += GEOS_FMT( "{:->{}}", "", cellSize );
+      }
+    }
+  }
+  topSeparator = GEOS_FMT( "+{:-<{}}+", "", sectionSeparator.size() - 2 );// -2 for ++
+}
+
+void Table::buildTitleRow( string & titleRows, string topSeparator, string sectionSeparator )
+{
+  titleRows = GEOS_FMT( "\n{}\n|", topSeparator );
+  titleRows +=  buildValueCell( Alignment::middle,
+                                tableTitle,
+                                (sectionSeparator.length() - 2) // -2 for ||
+                                );
+  titleRows += GEOS_FMT( "{}\n", "|" );
+}
+
+void Table::buildSectionRows( string sectionSeparator,
+                              string & rows,
+                              integer const nbRows,
+                              Section const section )
+{
+  for( integer idxRow = 0; idxRow< nbRows; idxRow++ )
+  {
+    rows += GEOS_FMT( "{:<{}}", "|", 1 +  borderMargin );
+    for( std::size_t idxColumn = 0; idxColumn < m_columns.size(); ++idxColumn )
+    {
+      string cell;
+
+      if( section == Section::header )
+      {
+        cell = m_columns[idxColumn].parameter.headerName[idxRow];
+      }
+      else
+      {
+        cell = m_columns[idxColumn].columnValues[idxRow];
+      }
+      integer cellSize = m_columns[idxColumn].m_maxStringSize.length();
+      rows += buildValueCell( m_columns[idxColumn].parameter.alignment,
+                              cell,
+                              cellSize );
+
+      if( idxColumn < m_columns.size() - 1 )
+      {
+        rows += GEOS_FMT( "{:^{}}", "|", columnMargin );
+      }
+
+    }
+    if( m_columns.size() == 1 )
+    {
+      rows +=  GEOS_FMT( "{:>{}}\n", "|", columnMargin );
+    }
+    else
+    {
+      rows += GEOS_FMT( "{:>{}}\n", "|", borderMargin + 1 );
+    }
+
+  }
+  if( nbRows != 0 )
+  {
+    rows += GEOS_FMT( "{}\n", sectionSeparator );
+  }
+}
+
+void Table::fillColumnsValuesFromCellsRows()
+{
+  for( size_t idxRow = 0; idxRow < m_cellsRows.size(); idxRow++ )
+  {
+    for( size_t idxColumn = 0; idxColumn < m_columns.size(); idxColumn++ )
+    {
+      m_columns[idxColumn].columnValues.push_back( m_cellsRows[idxRow][idxColumn] );
+    }
+  }
+}
+
+void Table::draw( std::ostream & oss )
+{
+  string tableOutput;
+  string rows;
+  string titleRows;
+  string topSeparator;
+  string sectionSeparator;
```

Comment:
In C++, don't hesitate to declare and define your parameters in the same lines, this way you can use the const keyword :
```c++
  string const tableOutput = titleRows + rows + '\n';
```